xls/simulation/cocotb/cocotb_xls.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 The XLS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cocotb
+
+from cocotb.triggers import RisingEdge
+from cocotb.binary import BinaryValue
+from cocotb.handle import SimHandleBase
+
+from cocotb_bus.drivers import BusDriver
+from cocotb_bus.monitors import BusMonitor
+
+from typing import Any, List
+
+
+class XLSChannelDriver(BusDriver):
+    _signals = ["data", "rdy", "vld"]
+
+    def __init__(
+        self, entity: SimHandleBase, name: str, clock: SimHandleBase, **kwargs: Any
+    ):
+        BusDriver.__init__(self, entity, name, clock, **kwargs)
+
+        self.bus.data.setimmediatevalue(0)
+        self.bus.vld.setimmediatevalue(0)
+
+    async def write(self, data: List[BinaryValue], sync: bool = True) -> None:
+        if sync:
+            await RisingEdge(self.clock)
+
+        for word in data:
+            self.bus.vld.value = 1
+            self.bus.data.value = word
+
+            while True:
```

Comment:
curious if it would be equivalent to `await RisingEdge(rdy)` here?

Reply:
It depends on whether the `rdy` signal is synchronous or not. If this is not guaranteed we should wait for a clock and then check the state of the `rdy` signal.

Reply:
> It depends on whether the `rdy` signal is synchronous or not.

I would think that this is the case, but maybe we can infer this property from generated verilog?

Reply:
Is there any reason why can't we use `while` which does not assume anything about the `rdy` signal?

Reply:
I looked again into the code. The `await RisingEdge(rdy)` is not equivalent to the current code, because there can be no changes of the `rdy` signal (for example when the receiving proc is able to consume all the data in a single burst). Instead we should check the value of the  `rdy` signal on every clock cycle.